### PR TITLE
chore: remove usage of lodash

### PR DESCRIPTION
**Pull request by talentlessguy**
All the lodash utils in the library can be replaced with pure JavaScript code. This PR removes lodash as a dependency while not changing any of the behavior. It's not using any latest ECMAScript features either, just regular ES6.

### CODE REVIEW

**Review by jlipps — COMMENTED** [review]
what do you think @mykola-mokhnach

File: lib/asyncbox.js [hunk 1]
```diff
@@ -206,11 +205,12 @@ async function asyncfilter (coll, filter, runInParallel = true) {
  */
 async function waitForCondition (condFn, options = {}) {
   /** @type {WaitForConditionOptions & {waitMs: number, intervalMs: number}} */
-  const opts = _.defaults(options, {
+  const opts = {
```
Comment (mykola-mokhnach) [inline lib/asyncbox.js:208]: _.defaults work slightly differently. I don't think this is the exact replacement

File: lib/asyncbox.js [hunk 2]
```diff
@@ -230,7 +230,7 @@ async function waitForCondition (condFn, options = {}) {
     }
     // if there is an error option, it is either a string message or an error itself
     throw error
-      ? (_.isString(error) ? new Error(error) : error)
+      ? (typeof error === 'string' ? new Error(error) : error)
```
Comment (mykola-mokhnach) [inline lib/asyncbox.js:233]: just checking typeof is not enough to verify a variable is a string.

**Comment by mykola-mokhnach** [comment]
I am not fully convinced about this PR, lodash is not obsolete and also replacing it might potentially introduce new issues if not done properly. 

From the pure Appium scope, where this lib is used in (almost) all other modules it also does not really help to achieve anything, like reducing the amount of dependencies or the overall installed package size.

File: lib/asyncbox.js (hunk 2 above)
Comment (talentlessguy) [reply lib/asyncbox.js:233]: how so? string is primitive type. template strings are still strings too

File: lib/asyncbox.js (hunk 1 above)
Comment (talentlessguy) [reply lib/asyncbox.js:208]: what's the difference between them?

File: lib/asyncbox.js (hunk 2 above)
Comment (talentlessguy) [reply lib/asyncbox.js:233]: if you need `Object#toString` support, I can add it, but doesn't seem like it accepts objects anyway. It's an either an `Error` object or a `string` in types:

https://github.com/jlipps/asyncbox/blob/ac95eb0d2722e2d6816980209f2586c3a94985ba/lib/asyncbox.js#L244-L251

**Comment by jlipps** [comment]
I guess I don't have strong feelings about this either. @talentlessguy can you share more about why you are proposing this change? Are you incorporating this package into something else and you don't want to see the lodash dep there, or what?

**Comment by talentlessguy** [comment]
@jlipps I don't personally depend on it but there are projects that do. This PR removes the need to uninstall lodash if you don't need it. The PR itself is really small, because lodash here is not needed at all and can be replaced with regular JavaScript. If this isn't a worthy improvement, feel free to close the PR.

**Pull request closed without merging**